sops/__init__.py

```diff
@@ -754,17 +773,18 @@ def walk_and_encrypt(branch, key, aad=b'', stash=None,
         if isinstance(v, dict):
             # recursively walk the tree
             branch[k] = walk_and_encrypt(v, key, aad=caad, stash=nstash,
-                                         digest=digest, isRoot=False)
+                                         digest=digest, isRoot=False, unencrypted=v_unencrypted)
         elif isinstance(v, list):
             branch[k] = walk_list_and_encrypt(v, key, aad=caad, stash=nstash,
-                                              digest=digest)
+                                              digest=digest, unencrypted=v_unencrypted)
         elif isinstance(v, ruamel.yaml.scalarstring.PreservedScalarString):
-            ev = encrypt(v, key, aad=caad, stash=nstash, digest=digest)
+            ev = encrypt(v, key, aad=caad, stash=nstash, digest=digest, unencrypted=v_unencrypted)
             branch[k] = ruamel.yaml.scalarstring.PreservedScalarString(ev)
         else:
-            branch[k] = encrypt(v, key, aad=caad, stash=nstash, digest=digest)
+            branch[k] = encrypt(v, key, aad=caad, stash=nstash, digest=digest, unencrypted=v_unencrypted)
```

Comment:
here the digest calculation doesn't depend on entering `encrypt()`, so let's not enter it at all.